RELEASE/scripts/autoscend/quests/level_11.ash

```diff
@@ -1323,15 +1355,18 @@ boolean L11_hiddenCity()
 		{
 			if(have_effect($effect[Thrice-Cursed]) == 0)
 			{
-				L11_hiddenTavernUnlock(true);
-				while(have_effect($effect[Thrice-Cursed]) == 0 && inebriety_left() >= $item[Cursed Punch].inebriety && canDrink($item[Cursed Punch]) && my_ascensions() == get_property("hiddenTavernUnlock").to_int() && !in_tcrs())
+				if(canDrinkCursedPunch && (inebriety_left() >= cursesNeeded*$item[Cursed Punch].inebriety))
 				{
-					buyUpTo(1, $item[Cursed Punch]);
-					if(item_amount($item[Cursed Punch]) == 0)
+					L11_hiddenTavernUnlock(true);
+					while(have_effect($effect[Thrice-Cursed]) == 0 && inebriety_left() >= $item[Cursed Punch].inebriety && my_ascensions() == get_property("hiddenTavernUnlock").to_int())
```

Comment:
a while loop here is just asking for infinite loops. those need a lot of safety mechanisms to make sure this does not happen. for handling every possible failure.
also it is not particularly necessary.

instead of a while loop it should first try to acquire as many cursed punches as needed and then multi drink them. (with safeties against over drinking)
less server hits and avoids the danger of a while loop

Reply:
the while loop is already in the code but seems easy to replace. the edited conditions should already protect against over drinking?